hail/src/main/scala/is/hail/methods/MatrixExportEntriesByCol.scala

```diff
@@ -69,8 +69,14 @@ case class MatrixExportEntriesByCol(parallelism: Int, path: String, bgzip: Boole
 
         val partFolder = partFileBase + partFile(d, i, TaskContext.get())
 
-        val fileHandles = Array.tabulate(endIdx - startIdx) { j =>
-          new OutputStreamWriter(bcFS.value.create(partFolder + "/" + j.toString + extension), "UTF-8")
+        val filePaths = Array.tabulate(endIdx - startIdx) { j =>
+          val finalPath = partFolder + "/" + j.toString + extension
+          val tempPath = bcFS.value.getTemporaryFile("file:///tmp", suffix=Some(extension))
```

Comment:
this is going to conflict with cotton's change I believe, will need to be extension=

Reply:
Actually, now it will be createTmpPath, which lives on ExecuteContext

Reply:
https://github.com/hail-is/hail/pull/8581/files#diff-8a2e24e243a3e3f841d27ef41a52a681R55

Reply:
will tag this as WIP, will be easier for me to rebase than him.

Reply:
thanks for the heads up